### PR TITLE
Morph

### PR DESCRIPTION
**Pull request by jietung**

### CODE REVIEW

File: src/main/java/seedu/exercise/logic/commands/AddCommand.java [hunk 1]
```diff
@@ -18,23 +18,22 @@
...
         + "Parameters: "
-        + PREFIX_NAME + "NAME "
-        + PREFIX_DATE + "PHONE "
-        + PREFIX_CALORIES + "EMAIL "
-        + PREFIX_QUANTITY + "ADDRESS "
+        + PREFIX_NAME + "EXERCISE NAME "
+        + PREFIX_DATE + "DATE "
+        + PREFIX_CALORIES + "CALORIES "
+        + PREFIX_QUANTITY + "QUANTITY "
         + "[" + PREFIX_MUSCLE + "TAG]...\n"
```
Comment (t-cheepeng) [inline src/main/java/seedu/exercise/logic/commands/AddCommand.java]: Here, it should be `MUSCLE` and not `TAG`

File: src/main/java/seedu/exercise/model/exercise/Property.java [hunk 2]
```diff
@@ -1,4 +1,4 @@
-package seedu.address.model.exercise;
+package seedu.exercise.model.exercise;
 
 /**
  * Represents a property of an {@code Exercise} object.
```
Comment (garylyp) [inline src/main/java/seedu/exercise/model/exercise/Property.java]: Is there a better way to make sure each individual property implement the following methods? For @weihaw08 to consider. Cos it seems like the Property abstract class is just forcing child classes to override the Object methods.

File: src/main/java/seedu/exercise/logic/commands/AddCommand.java [hunk 3]
```diff
@@ -18,23 +18,22 @@
...
+        + PREFIX_NAME + "Run "
+        + PREFIX_DATE + "22/09/2019 "
+        + PREFIX_CALORIES + "1500 "
+        + PREFIX_QUANTITY + "2.4 "
+        + PREFIX_MUSCLE + "Leg ";
 
-    public static final String MESSAGE_SUCCESS = "New person added: %1$s";
-    public static final String MESSAGE_DUPLICATE_PERSON = "This person already exists in the address book";
+    public static final String MESSAGE_SUCCESS = "New exercise added: %1$s";
+    public static final String MESSAGE_DUPLICATE_PERSON = "This exercise already exists in the exercise book";
```
Comment (t-cheepeng) [inline src/main/java/seedu/exercise/logic/commands/AddCommand.java]: Naming here should be `MESSAGE_DUPLICATE_EXERCISE`

File: src/main/java/seedu/exercise/logic/commands/AddCommand.java [hunk 4]
```diff
@@ -18,23 +18,22 @@
...
-        + PREFIX_DATE + "98765432 "
-        + PREFIX_CALORIES + "johnd@example.com "
-        + PREFIX_QUANTITY + "311, Clementi Ave 2, #02-25 "
-        + PREFIX_MUSCLE + "friends "
-        + PREFIX_MUSCLE + "owesMoney";
+        + PREFIX_NAME + "Run "
+        + PREFIX_DATE + "22/09/2019 "
+        + PREFIX_CALORIES + "1500 "
+        + PREFIX_QUANTITY + "2.4 "
+        + PREFIX_MUSCLE + "Leg ";
```
Comment (weihaw08) [inline src/main/java/seedu/exercise/logic/commands/AddCommand.java]: A whitespace not required at the end of the "Leg" string.

File: src/main/java/seedu/exercise/logic/parser/Prefix.java [hunk 5]
```diff
@@ -1,8 +1,8 @@
-package seedu.address.logic.parser;
+package seedu.exercise.logic.parser;
 
 /**
  * A prefix that marks the beginning of an argument in an arguments string.
- * E.g. 't/' in 'add James t/ friend'.
+ * E.g. 't/' in 'add Run t/ Leg'.
```
Comment (weihaw08) [inline src/main/java/seedu/exercise/logic/parser/Prefix.java]: May want to rewrite "t/" to "m/" to adhere to our prefixes.

File: src/main/java/seedu/exercise/model/exercise/Exercise.java [hunk 6]
```diff
@@ -1,16 +1,16 @@
...
 import java.util.HashSet;
 import java.util.Objects;
 import java.util.Set;
 
-import seedu.address.model.tag.Muscle;
+import seedu.exercise.model.tag.Muscle;
 
 /**
- * Represents a Person in the quantity book.
+ * Represents an Exercise in the quantity book.
```
Comment (weihaw08) [inline src/main/java/seedu/exercise/model/exercise/Exercise.java]: Need to change "quantity book" to "exercise book".

File: src/main/java/seedu/exercise/model/exercise/Quantity.java [hunk 7]
```diff
@@ -12,7 +12,7 @@
     public static final String MESSAGE_CONSTRAINTS = "Quantity can take any values, and it should not be blank";
 
     /*
-     * The first character of the address must not be a whitespace,
+     * The first character of the exercise must not be a whitespace,
```
Comment (weihaw08) [inline src/main/java/seedu/exercise/model/exercise/Quantity.java]: Should change "exercise" to "quantity"

File: src/main/java/seedu/exercise/logic/commands/AddCommand.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (jietung) [reply src/main/java/seedu/exercise/logic/commands/AddCommand.java]: done

File: src/main/java/seedu/exercise/logic/commands/EditCommand.java [hunk 8]
```diff
@@ -32,18 +32,18 @@
...
         + "Existing values will be overwritten by the input values.\n"
         + "Parameters: INDEX (must be a positive integer) "
         + "[" + PREFIX_NAME + "NAME] "
-        + "[" + PREFIX_DATE + "PHONE] "
-        + "[" + PREFIX_CALORIES + "EMAIL] "
-        + "[" + PREFIX_QUANTITY + "ADDRESS] "
+        + "[" + PREFIX_DATE + "DATE] "
+        + "[" + PREFIX_CALORIES + "CALORIES] "
+        + "[" + PREFIX_QUANTITY + "QUANTITY] "
         + "[" + PREFIX_MUSCLE + "TAG]...\n"
```
[the remark this answers is not in the document]
Comment (jietung) [reply src/main/java/seedu/exercise/logic/commands/EditCommand.java]: done

File: src/main/java/seedu/exercise/logic/commands/AddCommand.java (hunk 3 above)
Comment (jietung) [reply src/main/java/seedu/exercise/logic/commands/AddCommand.java]: done

File: src/main/java/seedu/exercise/logic/commands/AddCommand.java (hunk 4 above)
Comment (jietung) [reply src/main/java/seedu/exercise/logic/commands/AddCommand.java]: done

File: src/main/java/seedu/exercise/logic/parser/Prefix.java (hunk 5 above)
Comment (jietung) [reply src/main/java/seedu/exercise/logic/parser/Prefix.java]: done

File: src/main/java/seedu/exercise/model/exercise/Exercise.java (hunk 6 above)
Comment (jietung) [reply src/main/java/seedu/exercise/model/exercise/Exercise.java]: done

File: src/main/java/seedu/exercise/model/exercise/Quantity.java (hunk 7 above)
Comment (jietung) [reply src/main/java/seedu/exercise/model/exercise/Quantity.java]: done

File: src/main/resources/view/MainWindow.fxml [hunk 9]
```diff
@@ -14,7 +14,7 @@
 <fx:root minHeight="600" minWidth="450" onCloseRequest="#handleExit" title="Address App" type="javafx.stage.Stage"
```
Comment (kwekke) [inline src/main/resources/view/MainWindow.fxml]: May want to change the title

File: src/test/java/seedu/exercise/storage/JsonAdaptedExerciseTest.java [hunk 10]
```diff
@@ -1,21 +1,21 @@
...
-import seedu.address.model.exercise.Name;
-import seedu.address.model.exercise.Quantity;
+import seedu.exercise.commons.exceptions.IllegalValueException;
+import seedu.exercise.model.exercise.Calories;
+import seedu.exercise.model.exercise.Date;
+import seedu.exercise.model.exercise.Name;
+import seedu.exercise.model.exercise.Quantity;
 
 public class JsonAdaptedExerciseTest {
     private static final String INVALID_NAME = "R@chel";
```
Comment (kwekke) [inline src/test/java/seedu/exercise/storage/JsonAdaptedExerciseTest.java]: May want to change the values of these variables to better reflect their names.
eg "R@chel" to "D@nce"

File: src/test/java/seedu/exercise/logic/parser/AddCommandParserTest.java [hunk 11]
```diff
@@ -0,0 +1,174 @@
...
+            + QUANTITY_DESC_BASKETBALL + UNIT_DESC_BASKETBALL + MUSCLE_DESC_BASKETBALL,
+            new AddCommand(expectedExercise));
+
+        // multiple quantities - last quantity accepted
+        assertParseSuccess(parser, NAME_DESC_BASKETBALL + DATE_DESC_BASKETBALL
+            + CALORIES_DESC_BASKETBALL + QUANTITY_DESC_AEROBICS
+            + QUANTITY_DESC_BASKETBALL + UNIT_DESC_BASKETBALL + MUSCLE_DESC_BASKETBALL,
+            new AddCommand(expectedExercise));
+
+        // multiple quantities - last unit accepted
```
Comment (t-cheepeng) [inline src/test/java/seedu/exercise/logic/parser/AddCommandParserTest.java]: The comment here should be `multiple units - list unit accepted`

File: src/main/java/seedu/exercise/logic/commands/AddCommand.java [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+
+    public static final String COMMAND_WORD = "add";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds a exercise to the exercise list. "
+        + "Parameters: "
+        + PREFIX_NAME + "EXERCISE NAME "
+        + PREFIX_DATE + "DATE "
+        + PREFIX_CALORIES + "CALORIES "
+        + PREFIX_QUANTITY + "QUANTITY "
+        + "[" + PREFIX_MUSCLE + "MUSCLE]...\n"
```
Comment (t-cheepeng) [inline src/main/java/seedu/exercise/logic/commands/AddCommand.java]: The message printed out to feedback is lacking a units. Please add in `PREFIX_UNIT + "UNITS "`

File: src/test/java/seedu/exercise/logic/parser/AddCommandParserTest.java [hunk 13]
```diff
@@ -0,0 +1,174 @@
...
+
+        // missing date prefix
+        assertParseFailure(parser, NAME_DESC_BASKETBALL + VALID_DATE_BASKETBALL
+            + CALORIES_DESC_BASKETBALL + QUANTITY_DESC_BASKETBALL + UNIT_DESC_BASKETBALL, expectedMessage);
+
+        // missing calories prefix
+        assertParseFailure(parser, NAME_DESC_BASKETBALL + DATE_DESC_BASKETBALL
+            + VALID_CALORIES_BASKETBALL + QUANTITY_DESC_BASKETBALL + UNIT_DESC_BASKETBALL, expectedMessage);
+
+        // missing qauntity prefix
```
Comment (t-cheepeng) [inline src/test/java/seedu/exercise/logic/parser/AddCommandParserTest.java]: typo

File: src/test/java/seedu/exercise/storage/JsonAdaptedExerciseTest.java (hunk 10 above)
Comment (jietung) [reply src/test/java/seedu/exercise/storage/JsonAdaptedExerciseTest.java]: done

File: src/main/resources/view/MainWindow.fxml (hunk 9 above)
Comment (jietung) [reply src/main/resources/view/MainWindow.fxml]: done

File: src/test/java/seedu/exercise/logic/parser/AddCommandParserTest.java (hunk 11 above)
Comment (jietung) [reply src/test/java/seedu/exercise/logic/parser/AddCommandParserTest.java]: done

File: src/main/java/seedu/exercise/logic/commands/AddCommand.java (hunk 12 above)
Comment (jietung) [reply src/main/java/seedu/exercise/logic/commands/AddCommand.java]: done

File: src/test/java/seedu/exercise/logic/parser/AddCommandParserTest.java (hunk 13 above)
Comment (jietung) [reply src/test/java/seedu/exercise/logic/parser/AddCommandParserTest.java]: done

File: src/test/data/JsonExerciseBookStorageTest/invalidAndValidExerciseBook.json [hunk 14]
```diff
@@ -3,11 +3,13 @@
...
     "calories": "333",
-    "quantity": "20km"
+    "quantity": "20",
+    "unit" : "km"
   }, {
     "name": "Exercise with wrong date",
     "date": "33/20/2019",
     "calories": "234a",
-    "quantity": "50km"
+    "quantity": "50km",
```
[the remark this answers is not in the document]
Comment (jietung) [reply src/test/data/JsonExerciseBookStorageTest/invalidAndValidExerciseBook.json]: done

File: src/main/java/seedu/exercise/model/tag/Muscle.java [hunk 15]
```diff
@@ -1,7 +1,7 @@
-package seedu.address.model.tag;
+package seedu.exercise.model.tag;
```
[the remark this answers is not in the document]
Comment (jietung) [reply src/main/java/seedu/exercise/model/tag/Muscle.java]: will move over

**Pull request merged**